glib/src/thread_guard.rs

```diff
@@ -111,4 +112,40 @@ impl<T> Drop for ThreadGuard<T> {
     }
 }
 
+impl<T: Default> Default for ThreadGuard<T> {
+    fn default() -> Self {
+        Self::new(T::default())
+    }
+}
+
 unsafe impl<T> Send for ThreadGuard<T> {}
+
+impl<T: crate::ToValue + crate::StaticType> crate::ToValue for ThreadGuard<T> {
+    fn to_value(&self) -> crate::Value {
+        T::to_value(self.get_ref())
+    }
+
+    fn value_type(&self) -> crate::Type {
+        T::static_type()
+    }
+}
+
+unsafe impl<'a, T, C, E> crate::value::FromValue<'a> for ThreadGuard<T>
+where
+    T: crate::value::FromValue<'a, Checker = C> + crate::StaticType,
+    C: crate::value::ValueTypeChecker<Error = crate::value::ValueTypeMismatchOrNoneError<E>>,
+    E: std::error::Error + Send + Sized + 'static,
+{
+    type Checker = crate::value::ValueTypeOrNoneChecker<T, C, E>;
+
+    unsafe fn from_value(value: &'a crate::Value) -> Self {
+        match T::Checker::check(value) {
+            Err(crate::value::ValueTypeMismatchOrNoneError::UnexpectedNone) => panic!(),
```

Comment:
No idea if there's any point in checking the value since we can only panic really

Reply:
You need to check that the type matches and handle it accordingly. Just forward the checker result

Reply:
We are returning Self, so I think we can only unwrap it